Lib/test/test_exceptions.py

```diff
@@ -2105,6 +2105,18 @@ def test_range_of_offsets(self):
                         sys.__excepthook__(*sys.exc_info())
                     the_exception = exc
 
+    def test_encodings(self):
+        weird_text = '┬ó┬ó┬ó┬ó┬ó┬ó'
+        resolved_text = 'Γö¼├│Γö¼├│Γö¼├│Γö¼├│Γö¼├│Γö¼├│'
+        source = f"# -*- coding: cp437 -*-\nf(4, '{weird_text}' for x in range(1))"
+        try:
+            with open(TESTFN, 'w', encoding="utf-8") as testfile:
```

Comment:
Oh, the file I was using was actually written with encoding `cp437`. With this patch it results in the proper error:

```
  File "/tmp/test-cp437.py", line 2
    '┬ó┬ó┬ó┬ó┬ó┬ó' + f(4, 'Hi' for x in range(1)) # This line has a SyntaxError
                          ^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: Generator expression must be parenthesized
```

That seems simpler to test rather than this way?

Reply:
We still need to write to a file because this only triggers when reading source from files. Maybe I don't understand how you would like to simplify the test. Could you maybe write here a snippet?

Reply:
This is how I was imagining the test would look like:

```python
    def test_encodings(self):
        source = (
            '# -*- coding: cp437 -*-\n'
            '"┬ó┬ó┬ó┬ó┬ó┬ó" + f(4, x for x in range(1))\n'
        )
        try:
            with open(TESTFN, 'w', encoding='cp437') as testfile:
                testfile.write(source)
            rc, out, err = script_helper.assert_python_failure('-Wd', '-X', 'utf8', TESTFN)
            err = err.decode('utf-8').splitlines()

            self.assertEqual(err[-3], '    "┬ó┬ó┬ó┬ó┬ó┬ó" + f(4, x for x in range(1))')
            self.assertEqual(err[-2], '                          ^^^^^^^^^^^^^^^^^^^')
        finally:
            unlink(TESTFN)
```

Reply:
I have updated the test with your version :)